compiler/sigmatch.nim

```diff
@@ -900,6 +900,8 @@ proc typeRel(c: var TCandidate, f, aOrig: PType, doBind = true): TTypeRelation =
       if sameDistinctTypes(f, a): result = isEqual
       elif f.base.kind == tyAnything: result = isGeneric
       elif c.coerceDistincts: result = typeRel(c, f.base, a)
+    elif a.kind == tyNil and f.base.kind in NilableTypes and tfNotNil notin f.base.flags:
+      result = isEqual
```

Comment:
``result = f.allowsNil`` for consistency with how ``nil`` it dealt with in the other cases.